internal/redact/redact.go

```diff
@@ -33,3 +36,77 @@ func FromContext(ctx context.Context) (bool, string) {
 	reason, ok := ctx.Value(redactKey).(string)
 	return ok, reason
 }
+
+// Error redacts potentially sensitive query parameter values in the URL from the error's message.
+func Error(err error) error {
```

Comment:
-1 on this because we throw away error types, can you do this in here instead?

https://github.com/google/go-containerregistry/blob/d187a716b559771a3768caa9c4dba796cb4650b3/pkg/v1/remote/transport/error.go#L67-L87

Reply:
Or redact this: https://pkg.go.dev/net/url#Error